Automattic/sensei
# Allow styling lesson action buttons in learning mode

Resolves https://github.com/Automattic/sensei/issues/7088

## Proposed Changes
Enabled using Lesson Actions block in learning mode.

In learning mode, we were using the learning mode-specific `course-theme-lesson-actions` block. This block didn't allow many customizations, especially lacked button customizability which resulted in many users facing issues with styling it as it was possible to style them only using CSS.

Here we have made some changes to the Lesson Actions block and used that instead in learning mode. it has most of the customizability and behavior of buttons. Based on the number of ZD tickets in the reference comments of the issue, looks like this is the number one most awaited feature improvement :p

Note: We don't have the Checked icon in the Completed button and Right arrow in next lesson now. As SVG is not supported as inline image inside buttons. We could programmatically add them, but we want these buttons to be fully customizable by the user and keep the buttons as close to Core buttons as possible.

## Testing Instructions


1. Make sure you do not have any saved version of the (Learning Mode - Default) template in site editor. otherwise you will not find these changes.
2. Create a course with many lessons, some with quizzes, some without.
3. Take some quizzes, complete some lessons, keep some quizzes awaiting grade, complete some quizzes.
4. Make sure in each of those scenarios, the button colors, shapes and texts match the previous LM buttons (except for dynamic color changes, because you can't do it from block editor)
5. Check with different themes, block and non block themes.
6. Check in mobile view.
7. Check without learning mode as well.

#### Screenshots
Course Theme:
![Screenshot 2023-10-19 at 8 41 42 AM](https://github.com/Automattic/sensei/assets/6820724/f0940823-32ae-4921-a681-9df208550a34)

Color customized in Divi:
![Screenshot 2023-10-19 at 8 43 39 AM](https://github.com/Automattic/sensei/assets/6820724/ec5a3c40-e7eb-4643-8ed6-428b8967bf67)

Mobile view:
![Screenshot 2023-10-19 at 8 45 17 AM](https://github.com/Automattic/sensei/assets/6820724/ca36e4e7-49d7-4f99-90bf-2f2b2f50cbd7)

## Pre-Merge Checklist
